utils/core/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -241,7 +241,12 @@ fun OkHttpClient.download(url: String, acceptEncoding: String? = null): Result<P
         val body = response.body
 
         if (!response.isSuccessful || body == null) {
-            throw HttpDownloadError(response.code, response.message)
+            
+            val code=response.code 
+            val message=response.message
+            response.close()
```

Comment:
As [according to OkHttp docs](https://github.com/square/okhttp/blob/0a27e9ff30c6129f4fdeb1f99c07c9b009bcc537/okhttp/src/main/kotlin/okhttp3/Response.kt#L295-L303) `close()` on a `Response` does nothing else than calling `body().close()` we IMO should

1. not need to save the `code` and `message` before closing,
2. verify that the `body` is not null to not run into the check inside OkHttp (which throws an exception).

So, could you verify whether simply adding the line `body?.close()` would also work?

Reply:
Note: Will also fix the detect issues.:)

Reply:
Tested suggest solution IRL project (and no longer get the leak messages on failed downloads), updated PR, fixed detekt issues.

Reply:
Thanks. Would you mind squashing your two commits into one, and maybe change the commit message to say "failure" instead of "fail"?